### ISSUE TITLE
Have a field for time

### ISSUE THREAD

**Issue by ealmdahl** [hyperoslo/hyper-recipes#11]
We should have a field for how much time each recipe would take, in minutes. Like
`time: 45`

**Comment by jgorset** [comment]
That's a great example you've got there.

**Comment by 3lvis** [comment]
I don't really find a good use of having time when following recipes, it's subjective. I prefer to have difficulty levels.

**Comment by sindrenm** [comment]
Are you talking about a time to make the meal in total, @ealmdahl, or more specific like “fry the chicken for 2.5 minutes on both sides” and “boil the pasta for 30 minutes”?

**Comment by ealmdahl** [comment]
@sindrenm, I'm thinking estimated total time.

Well, @3lvis, I beg to differ. Difficulty might be subjective, an experienced baker might think a level 3 cake is still fairly easy, but a baby still takes 9 months to bake. Also, if you're in a rush, like if you forgot to bake something for the parents' meeting at school, you don't make something that needs 3 hours in the oven even though it practically makes itself.

**Comment by sindrenm** [comment]
I think we should have both time to make the meal **and** a difficulty level.

**Comment by 3lvis** [comment]
What's the point on having the most complex API ever? This should be super simple.

We don't plan to make a fully featured recipes app.

**Comment by jgorset** [comment]
Wait wait wait – our goal with this _isn't_ to release several dozen recipes applications on the App Store?

**Comment by ealmdahl** [comment]
No. But I would like the API to match this design sketch, which I think is a nice base for discussion with the applicants. 
![design sketch](https://cloud.githubusercontent.com/assets/6986854/7748560/f5d74fd2-ffc5-11e4-9c83-b2cd2bd63b67.png)

Also, I think it really makes sense to have both a time and the ingredients and instruction arrays. It has more to do with making a realistic API than the most complex ever.

**Comment by 3lvis** [comment]
@ealmdahl I don't know if we share this, but the main idea of HyperRecipes is not to make a fully featured app, that's easy, and it's not a challenge at all. The idea is to try collaboration and how does the developer structure his code.

**Comment by ealmdahl** [comment]
indeed, @3lvis, but does it hurt to to have a better structured API? I think not.
